extras/eof_test.go

```diff
@@ -0,0 +1,264 @@
+	client, err := posthog.NewWithConfig(
+		"test-api-key",
+		posthog.Config{
+			Endpoint: url,
+			Transport: &timeoutTransport{
+				rt:      http.DefaultTransport,
+				timeout: 60 * time.Second,
+			},
+			Interval:  100 * time.Millisecond, // Fast flush for testing
+			BatchSize: 1,                      // Send immediately after 1 event
+			Logger:    posthog.StdLogger(log.New(os.Stderr, "[posthog] ", log.LstdFlags), true),
+		},
+	)
+	require.NoError(t, err, "Failed to create PostHog client")
+
+	// Enqueue an event - this will trigger a batch send to the mock server
+	err = client.Enqueue(posthog.Capture{
+		DistinctId: "testuser",
+		Event:      "test event",
+		Properties: map[string]interface{}{
+			"hello": "world",
+		},
+	})
+	require.NoError(t, err, "Failed to enqueue event")
+	t.Log("Event enqueued, waiting for batch send...")
+
+	// Wait for the batch to be sent and processed
+	time.Sleep(500 * time.Millisecond)
+
+	// Close the client - this will flush any remaining events
+	err = client.Close()
+	if err != nil {
+		t.Logf("Client close error (may indicate EOF error): %v", err)
+	}
+
+	t.Logf("Server received %d request(s)", server.RequestCount())
+}
+
+func TestPostHogHeadersOnlyCloseWithCallback(t *testing.T) {
+	server := flakyhttp.NewServer(flakyhttp.Config{
+		Scenario:        flakyhttp.ScenarioHeadersOnlyClose,
+		PartialBodySize: 0, // Send 0 bytes of body (promised 1000) - forces EOF on first read
+	})
+
+	url, err := server.Start()
+	require.NoError(t, err, "Failed to start mock server")
+	defer server.Close()
+	t.Logf("Mock server started at %s", url)
+
+	// Channel to capture callback errors
+	errChan := make(chan error, 10)
+
+	// Create PostHog client with callback to capture errors
+	client, err := posthog.NewWithConfig(
+		"test-api-key",
+		posthog.Config{
+			Endpoint: url,
+			Transport: &timeoutTransport{
+				rt:      http.DefaultTransport,
+				timeout: 60 * time.Second,
+			},
+			Interval:  100 * time.Millisecond,
+			BatchSize: 1,
+			Logger:    posthog.StdLogger(log.New(os.Stderr, "[posthog] ", log.LstdFlags), true),
+			Callback:  &testCallback{t: t, errChan: errChan},
+		},
+	)
+	require.NoError(t, err, "Failed to create PostHog client")
+
+	// Enqueue an event
+	err = client.Enqueue(posthog.Capture{
+		DistinctId: "testuser",
+		Event:      "test event",
+		Properties: map[string]interface{}{
+			"hello": "world",
+		},
+	})
+	require.NoError(t, err, "Failed to enqueue event")
+	t.Log("Event enqueued, waiting for callback...")
+
+	// Wait for callback with timeout
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+	defer cancel()
+
+	select {
+	case capturedErr := <-errChan:
+		t.Logf("Captured error via callback: %v", capturedErr)
+	case <-ctx.Done():
+		t.Log("No error captured within timeout (PostHog client may have handled it)")
+	}
+
+	client.Close()
+	t.Logf("Server received %d request(s)", server.RequestCount())
```

Comment:
Shouldn't this test assert something? I see it logging info, but the test itself will always pass because there are no assertions.